src/Native/Runtime/thread.cpp

```diff
@@ -1295,9 +1295,15 @@ EXTERN_C NOINLINE void FASTCALL RhpReversePInvokeAttachOrTrapThread2(ReversePInv
 // PInvoke
 //
 
+extern "C" int InitializeRuntime();
```

Comment:
This creates a static dependency from Runtime to bootstrapper. These dependencies are problematic. (E.g. for .NET Native on Windows UWP, we build runtime as separate .dll that does not include bootstrapper.)

Can InitializeRuntime here be a function pointer that the bootstrapper sets - e.g. in a static constructor?

Reply:
@jkotas sounds reasonable. Is there a part in the code where something similar is done? I might be able to get some ideas from there

Reply:
I do not think there is anything like that in CoreRT today. There are number of examples out there on how to achieve it: https://www.bing.com/search?q=__attribute__%20((constructor))